DMCompiler/Compiler/DMPreprocessor/DMPreprocessor.cs

```diff
@@ -135,12 +142,20 @@ public sealed class DMPreprocessor : IEnumerable<Token> {
                         if (!_lastIfEvaluations.TryPop(out var _))
                             DMCompiler.Emit(WarningCode.BadDirective, token.Location, "Unexpected #endif");
                         break;
-                    case TokenType.DM_Preproc_Identifier:
-                        _currentLineWhitespaceOnly = false;
-                        if(!TryMacro(token)) {
-                            yield return token;
+                    case TokenType.DM_Preproc_Identifier: {
+                        if (TryMacro(token)) {
+                            break;
+                        }
+
+                        while (_bufferedWhitespace.TryPop(out var whitespace)) {
+                            yield return whitespace;
                         }
+                        _currentLineContainsNonWhitespace = true;
+                        _canUseDirective = false;
+
+                        yield return token;
                         break;
```

Comment:
Could fall-through to one of the cases below to remove the duplicated code
```suggestion
                        // Otherwise treat it like any other normal token
                        goto case TokenType.DM_Preproc_Punctuator;
```